test/framework/src/main/java/org/opensearch/test/ParameterizedOpenSearchIntegTestCase.java

```diff
@@ -44,4 +46,14 @@ public void afterTests() {
         dynamicSettings.keySet().forEach(settingsToUnset::putNull);
         client().admin().cluster().prepareUpdateSettings().setPersistentSettings(settingsToUnset).get();
     }
+
+    public void indexRandomForConcurrentSearch(int segments, String... indices) throws InterruptedException {
+        if (dynamicSettings.get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey()).equals("true")) {
+            indexRandomForMultipleSlices(segments, indices);
+        }
+    }
+
+    public void indexRandomForConcurrentSearch(String... indices) throws InterruptedException {
+        indexRandomForConcurrentSearch(2, indices);
```

Comment:
these functions rely on `dynamicSettings` and used for the parameterized tests only. Hence, i thought of keeping these functions in this class. thoughts?

Reply:
For dynamic settings, these functions can fetch it from cluster setting using `client().admin().cluster()...`. That way I see it more like a generic utility nothing specific to parameterization. That way any tests should be able to use these utility as needed.

Reply:
the way to fetch cluster settings in the class is via internalCluster like - `internalCluster().clusterService().getClusterSettings()` - since internal cluster is already loaded before the parameterization, it doesn't contain the updated value of concurrent search done by the parameterization. Hence, keeping it in the parameterized class only.

Reply:
I would expect parameterization to update the cluster setting before running the tests. The utility function is called from each test so it should get updated settings. You may need to use the mechanism I shared above to explicitly make the call if the method you shared is fetching the cached value for some reason. But for now I am fine with keeping it here as well.